models/articles-models.js

```diff
@@ -1,37 +1,78 @@
 const db = require("../db/connection");
 
-exports.fetchAllArticles = async (sort = "created_at", order = "desc") => {
-  const validOrders = ["asc", "ASC", "desc", "DESC"];
+exports.fetchAllArticles = async (
+  sort = "created_at",
+  order = "desc",
+  topic
+) => {
+  // pulling valid topics from topics table
+  const { rows: topics } = await db.query(`SELECT slug FROM topics`);
```

Comment:
I like what you're doing here - but this could be another use case for `checkIfExists` - you would only need to check the existence of a topic if the array of articles came back empty